### ISSUE TITLE
This package causes App Engine deployments to fail.

### ISSUE THREAD

**Issue by flexgrip** [laravelista/comments#47]
Hello,

This package is awesome. I set it up in a few minutes, adjusted it to my liking and then tried to deploy a test version to my appengine production project.

As far as I can tell its happening during the composer scripts. As an example:

```
Step #1 - "builder": In Compiler.php line 36:
Step #1 - "builder": 
Step #1 - "builder": Please provide a valid cache path. 

... then ...

Step #1 - "builder": Generating optimized autoload files
Step #1 - "builder": > Illuminate\Foundation\ComposerScripts::postAutoloadDump
Step #1 - "builder": > @php artisan package:discover --ansi
Step #1 - "builder": Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1
Step #1 - "builder": 
Step #1 - "builder": error: `composer_install` returned code: 1
Step #1 - "builder": Traceback (most recent call last):
Step #1 - "builder": File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
Step #1 - "builder": "__main__", fname, loader, pkg_name)
Step #1 - "builder": File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
Step #1 - "builder": exec code in run_globals
Step #1 - "builder": File "/usr/local/bin/ftl.par/__main__.py", line 65, in <module>
Step #1 - "builder": File "/usr/local/bin/ftl.par/__main__.py", line 57, in main
Step #1 - "builder": File "/usr/local/bin/ftl.par/__main__/ftl/common/ftl_error.py", line 58, in UserErrorHandler
Step #1 - "builder": IOError: [Errno 2] No such file or directory: '""/output'
Finished Step #1 - "builder"
ERROR
ERROR: build step 1 "gcr.io/gae-runtimes/php72_app_builder:php72_20190515_7_2_17_RC00" failed: exit status 1
```

Just wanted to know if you had any idea what could be causing this? I am using Laravel 5.7, php72 and if I `composer remove` your package, the deployment scripts run fine. My suspicion is that it is trying to cache or discover something in your package (or the included markdown parser) and failing.

Keep in mind, I realize your package probably works on 99% of deployments... and appengine is a weird case. So I'm just looking for advice if you have time. The issue seems to happen at the same step as this previously closed ticket: https://github.com/laravelista/comments/issues/16

Thanks!

**Comment by mabasic** [comment]
Hi, I'm glad that you like it.

I don't know what may be causing your issue. The package does not use or depend on any cache...

I have a new version (not yet ready for a release) in the 'develop' branch. Maybe that release will work for you. I have made a few changes there and plan to implement more changes.

Do you want me to ping you when it will be ready? My guess is this weekend.

**Comment by flexgrip** [comment]
Absolutely. I plan on adding a bunch if things to it like rate limiting (to keep the trolls at bay) and I was going to see if I could auto update the comments via pusher. Once done, I’ll send a PR. Hopefully it’ll fit your intended purpose and coding style so it can be useful to you.

**Comment by mabasic** [comment]
That would be nice. Thank you

**Comment by mabasic** [comment]
Just an update.

I haven't had the time the last weekend for this package, but I am working on creating a blog that will use this package and I think that once I start implementing comments in it, is when I will update this package. I'll let you know.

**Comment by flexgrip** [comment]
Appreciate it. I am going to be taking a crack at getting these comments to work again this week. I really don't want to make my own. But I need to figure out exactly which dep is causing the deployment issue. Unfortunately, the logs are huge and messy during appengine deployments. And it doesn't give me much insight into exactly what is failing. 

Was also thinking of upgrading from L5.7 to 5.8 with the latest version of Spark. But we'll see. I'm sure it will be a pain.

**Comment by drionix** [comment]
After almost a month of pulling out my hair because i don't know what caused error (i have lots of other package installed too), I finally found the problem on my own and only after that I found this page.

The issue is not the caused by this package itself, but rather it seems like the problem with combined implementation of this package, laravel and App Engine lol

In short, I could confirm the particular line that caused this issue is

```
Blade::component('comments::components.comments', 'comments');
```
 
I don't really know what happen behind the scene, but I guess laravel cache blade components, and App Engine tried to do it in the "cloud builder", which is not a prod/dev environment, so it cannot access laravel files, therefore the error.

The solution? I did this and it works

```
if (!\App::runningInConsole()){
	Blade::component('comments::components.comments', 'comments');
}
```

**Comment by flexgrip** [comment]
How cow @drionix. You are a beast. I couldn’t even find detailed enough logs to trace it down.

Also, kudos to you. It sounds like you’re in appengine standard env? I’m working on getting memcached and other services working with L5.8. PM me if you’ve got any ideas.

Thanks again!

**Comment by mabasic** [comment]
Maybe this will help you: https://cloud.google.com/community/tutorials/run-laravel-on-appengine-standard

See the part where it says `APP_STORAGE` and `VIEW_COMPILED_PATH`.

Also check the `beyondcode/laravel-dump-server` removal step.

It seems to me that laravel will cache views and logs and appengine does not allow it to write to /storage. Let me know if these steps work for you.

**Comment by drionix** [comment]
> Maybe this will help you: https://cloud.google.com/community/tutorials/run-laravel-on-appengine-standard
> 
> See the part where it says `APP_STORAGE` and `VIEW_COMPILED_PATH`.
> 
> Also check the `beyondcode/laravel-dump-server` removal step.
> 
> It seems to me that laravel will cache views and logs and appengine does not allow it to write to /storage. Let me know if these steps work for you.

Actually, I build my web based on that tutorial. Everything was fine until around 2 months ago. However, I only noticed the error about 1 month ago since app engine always cache what it builds.

As this package has no 'suspicious' commits in the last 1 month, I know that the problem must be from laravel or app engine.

And like I said previously, somehow I found out that  laravel seems to cache blade components, and App Engine Cloud Build tried to parse it in the "cloud build" environment, which probably is a separate environment (not in a prod/dev environment), so it cannot access laravel files and therefore the error.

In conclusion, it is not the problem with this package. However, it would be awesome if you could help mitigate this issue by modifying 
```
Blade::component('comments::components.comments', 'comments');
```
in `ServiceProvider.php` to
```
if (!\App::runningInConsole()){
	Blade::component('comments::components.comments', 'comments');
}
```

**Comment by mabasic** [comment]
@drionix Thank you for explaining, but I don't like adding this code to my package to patch this specific issue.

Currently I am working on a new version in which I have replaced:

```
Blade::component('comments::components.comments', 'comments');
```

with

```
Blade::include('comments::components.comments', 'comments');
```

Can you try using the package with that change and see if it works for you then?

**Comment by mabasic** [comment]
Try the latest release: https://github.com/laravelista/comments/releases/tag/3.0.0
